packages/dds/tree/src/simple-tree/tree.ts

```diff
@@ -165,29 +251,95 @@ export interface TreeView<TSchema extends ImplicitFieldSchema> extends IDisposab
 	 */
 	upgradeSchema(): void;
 
+	/**
+	 * Initialize the tree, setting the stored schema to match this view's schema and setting the tree content.
+	 *
+	 * Only valid to call when this view's {@link SchemaCompatibilityStatus.canInitialize} is true.
+	 *
+	 * Applications should typically call this function before attaching a `SharedTree`.
+	 * @param content - The content to initialize the tree with.
+	 */
+	initialize(content: InsertableTreeFieldFromImplicitField<TSchema>): void;
+
 	/**
 	 * Events for the tree.
 	 */
 	readonly events: Listenable<TreeViewEvents>;
 }
 
 /**
- * Information about how a view schema was incompatible.
+ * Information about a view schema's compatibility with the document's stored schema.
+ *
+ * See SharedTree's README for more information about choosing a compatibility policy.
  * @public
  */
-export interface SchemaIncompatible {
+export interface SchemaCompatibilityStatus {
 	/**
-	 * True iff the view schema supports all possible documents permitted by the stored schema.
+	 * Whether the view schema allows exactly the same set of documents as the stored schema.
+	 *
+	 * @remarks
+	 * Equivalence here is defined in terms of allowed documents because there are some degenerate cases where schemas are not
+	 * equal in a strict sense but still allow the same documents, and the document notion is more useful to applications.
+	 *
+	 * Examples which are expressible where this may occur include:
+	 * - schema repository `A` has extra schema which schema `B` doesn't have, but they are unused (i.e. not reachable from the root schema)
+	 * - field in schema `A` has allowed field members which the corresponding field in schema `B` does not have, but those types are not constructible (ex: an object node type containing a required field with no allowed types)
+	 *
+	 * These cases are typically not interesting to applications.
+	 */
+	readonly isExactMatch: boolean;
```

Comment:
I don't think this is really the best way to express this.

Stored schema and view schema have different types, store different in information and have different degrees of freedom, so you can't compare them for equivalence.

I think "fullyCompatible" might better convey the semantics here? That might not be a great name once we have adapters which can provide full compatibility if you want this to be false in that case though?

Maybe there is some better term.

Reply:
A lot of this feels so far removed from the public API that I'd rather lose precision against how the API fits into our layering for clarity in the API's semantics. I don't really like using some compatibility-based term for precisely the reason you mention, but if we make a point to clearly define 'fully compatible' and its distinction from 'compatible' I could probably be convinced.

Exact naming of this field is less of a concern to me than your points on stored schema and view schema discrepancy though. Can you give an example where the distinctions between stored and view schema materialize in a way that makes the API surface around compatibility awkward? I've been operating under the assumption that because we derive stored schema transparently from view schema (from the perspective of the app), that concepts like this can be defined pretty sensibly in terms of allowed documents.

Reply:
Brought this up in API council--how do you feel about "areEquivalent"?

Reply:
I went ahead and updated the PR to `isEquivalent` rather than `isExactMatch`.